connect_plants_lines.py

```diff
@@ -0,0 +1,129 @@
+    processed = {}
+    for feature in feature_list:
+        geometry = feature['geometry']['coordinates']
+        if len(geometry) > 1:
+            if geometry[0][0] == geometry[1][0]:
+                geometry = [geometry[1][::-1] + geometry[0]] + geometry[
+                                                               2:]  # merges lines together if one section is an extension of another one
+            elif geometry[0][-1] == geometry[1][0]:
+                geometry = [geometry[0] + geometry[1]] + geometry[2:]
+            elif geometry[0][0] == geometry[1][-1]:
+                geometry = [geometry[1] + geometry[0]] + geometry[2:]
+            elif geometry[0][-1] == geometry[1][-1]:
+                geometry = [geometry[0] + geometry[1][::-1]] + geometry[2:]
+
+        info = feature['properties']
+        processed[info['ID']] = [geometry[-1][-1], geometry[0][0]]  # adds new power line to list of processed objects
+    return processed
+
+
+def find_matching_plants(point, line_id, es):             # matches power plants and lines
+    global DISTANCES
+    results = es.search(index="powerplants", body={"query": {
+        "bool": {
+            "must": {
+                "match_all": {}
+            },
+            "should": {
+                "distance_feature": {
+                    "field": "location",
+                    "pivot": "5000km",                   # large distance to guarantee at least one match, closest points first on return
+                    "origin": point                      # point is a long-lat array
+                }
+            }
+        }
+    }})
+
+    if len(results['hits']['hits']):
+        plant_id = results['hits']['hits'][0]['_source']['gppd_idnr']
+        plant_location = results['hits']['hits'][0]['_source']['location']
+        distance1 = calc_distance(plant_location[0], plant_location[1], point[0], point[1])
+        DISTANCES.append({'line_id': line_id, 'plant_id': plant_id, 'distance': distance1})   #global distances dictionary
+        return (plant_id, plant_location)
+
+
+def make_line_plant_connections(power_line_data, elastic_search):
+    connections = {}
+
+    for id in power_line_data:
+        start_of_pline = power_line_data[id][0]
+        end_of_pline = power_line_data[id][0]
+        connections[id] = {}
+        connections[id]['first'] = find_matching_plants(start_of_pline, id, elastic_search)
+        connections[id]['second'] = find_matching_plants(end_of_pline, id, elastic_search)
+
+    return connections
+
+
+loaded_data = get_pline_data()
+elastic_connection = prepare_elasticsearch()
+print('retrieved and loaded')
+processed_data = process_lines(loaded_data)
+connections_dictionary = make_line_plant_connections(processed_data, elastic_connection)
+
+bulk_connections_list = ({
+    '_index': connections_index_name,
+    '_ID': position,
+    '_source': connection} for position, connection in enumerate(DISTANCES)
+)
+
```

Comment:
I think I see now where the global DISTANCES is coming in.   I think this can be simplified to not use a global and just return the completed dictionary.  Consider if the json list can be built directly as the data is being processed to avoid the intermediate dictionary.